src/extension/background-script/CryptoServices/utils.ts

```diff
@@ -28,22 +28,41 @@ export interface TypedMessageText extends TypedMessageMetadata {
     readonly type: 'text'
     readonly content: string
 }
+export interface TypedMessageImage extends TypedMessageMetadata {
+    readonly type: 'image'
+    readonly content: string
+}
 export interface TypedMessageComplex extends TypedMessageMetadata {
     readonly type: 'complex'
     readonly items: readonly TypedMessage[]
 }
 export interface TypedMessageUnknown extends TypedMessageMetadata {
     readonly type: 'unknown'
 }
-export type TypedMessage = TypedMessageText | TypedMessageComplex | TypedMessageUnknown
-export function makeTypedMessage(text: string, meta?: ReadonlyMap<string, unknown>): TypedMessageText
-export function makeTypedMessage(content: string, meta?: ReadonlyMap<string, unknown>): TypedMessage {
-    if (typeof content === 'string') {
-        const text: TypedMessageText = { type: 'text', content, version: 1, meta }
-        return text
-    }
-    const msg: TypedMessageUnknown = { type: 'unknown', version: 1, meta }
-    return msg
+export type TypedMessage = TypedMessageText | TypedMessageImage | TypedMessageComplex | TypedMessageUnknown
+export function makeTypedMessage(
+    type: 'text' | 'image' | 'complex' | 'unknown',
```

Comment:
I prefer a factory function for each type cause their construction argument may variant a lot.